docs/git/index.md

```diff
@@ -0,0 +1,220 @@
+* Add labels to the pull request (e.g. marking it as complete, or unfinished).
+
+If they give you suggestions, you should try to implement the suggestions, and push a new commit. If you're confused, struggling, or find them unclear, you can respond to the comment with a comment of your own, or ask on Slack.
+
+## History
+
+One useful features of Git is how it stores your commits. In Github Desktop, if you open the History tab, you can see a list of each commit that's been made in the repository, with the oldest at the bottom and the newest at the top.
+
+If you click on one of the commits, you'll see the changes that happened in it. This can be really useful to understand how the repository evolved to how it looks today. It can also help us to find out when bugs were introduced!
+
+You may notice that your commits all have messages like "Update README.md", whereas the ones before you started editing have different messages.
+
+These messages can be really useful to understand what a change did without having to read the whole thing. Let's try out an exercise to help us understand this:
+
+### Exercise (10 minutes)
+
+1. Clone https://github.com/illicitonion/git-log-example TODO: Move this repo into CodeYourFuture. TODO: Give instructions for how to clone a second repo (beacuse the screen is different from your first time).
+2. Have a read of the README.md file. See if you can find a problem in the file.
+3. Look through the history in Github Desktop. See if you can work out when and why the problem was introduced.
+4. Make a pull request fixing the problem.
+
+## Commit messages
+
+If you looked through the history of the Git repository, you may have been able to see where the problem came in just from the commit message, without having to look at the changes that were actually made. If the commit messages were all just "Update README.md", that would've been much harder!
+
+When we make commits in Git, we try to give clear, helpful messages, describing what changed, and why. (The why is really important! You can always work out _what_ changed by reading the change itself, but it's much harder to work out _why_ if no one wrote it down!)
+
+There is a convention when using Git to use the following format for commit messages:
+
+```
+One sentence summary
+
+Longer explanation of how, the reasons, exceptions, or anything that may be surprising.
+This may be many sentences, and keep going for a long time.
+```
+
+Try to use this format when making your own commits. Github Desktop tries to encourage this by having two boxes above the "Commit to main" button - one which is just one line, and one where you can put lots of lines.
+
+TODO: Insert image of a good commit message in Github Desktop.
+
+## Homework going forwards
+
+For the next three modules, most weeks your homework is going to be to clone a repository from Github, write some code, and make a Pull Request with the code.
+
+## Glossary
```

Comment:
From @40thieves:

At the risk of signing myself up for more work, what do you think of making a cheatsheet for Git stuff? A glossary is the sort of thing I would put on that.

Reply:
Love it! This also ties in really nicely with @nbogie's [suggestion](https://github.com/CodeYourFuture/syllabus/issues/64#issuecomment-711456885) that this class be centred around a cheatsheet/reference, and be viewed as an introduction to that, rather than being a standalone class which happens to have a cheatsheet.